File: tests/fixtures/sample-python-project/services/user_service/services/user_service.py
from ..repositories.user_repository import UserRepository

repo = UserRepository()


class UserService:
    def get_all(self) -> list:
        return repo.find_all()

    def get_by_id(self, user_id: str):
        return repo.find_by_id(user_id)

    def create(self, data: dict):
        return repo.create(data)

    def delete(self, user_id: str) -> None:
        return repo.delete(user_id)

File: tests/fixtures/sample-csharp-project-negative/services/UserService/Services/UserService.cs
using UserServiceApp.Repositories;

namespace UserServiceApp.Services;

public class UserService
{
    private readonly UserRepository _repo;

    public UserService()
    {
        _repo = new UserRepository();
    }

    public List<Models.User> GetAll()
    {
        return _repo.FindAll();
    }

    public Models.User? GetById(string userId)
    {
        return _repo.FindById(userId);
    }

    public Dictionary<string, object> Create(Dictionary<string, object> data)
    {
        var user = _repo.Create(data);
        return new Dictionary<string, object>
        {
            ["id"] = user.Id,
            ["name"] = user.Name!,
            ["email"] = user.Email
        };
    }

    public void Delete(string userId)
    {
        _repo.Delete(userId);
    }
}
